tag_invoke cannot be used within the cncr namespace

**Describe the bug**
The `tag_invoke` niebloid is currently defined directly within the `cncr` namespace. This prevents the definition of `tag_invoke` based CPOs within that namespace.
```
/root/.vs/concrete/src/dplx/cncr/tag_invoke.test.cpp:67:24: error: redefinition of 'tag_invoke' as different kind of symbol
    friend inline auto tag_invoke(cncr_tests::ti_test_cp_fn,
                       ^
/root/.vs/concrete/src/dplx/cncr/tag_invoke.hpp:50:45: note: previous definition is here
inline constexpr detail::cpo::tag_invoke_fn tag_invoke{};
                                            ^
/root/.vs/concrete/src/dplx/cncr/tag_invoke.test.cpp:88:6: error: redefinition of 'tag_invoke' as different kind of symbol
auto tag_invoke(ti_test_cp_fn, cncr_tests::ti_test_type &self, std::size_t n)
     ^
/root/.vs/concrete/src/dplx/cncr/tag_invoke.hpp:50:45: note: previous definition is here
inline constexpr detail::cpo::tag_invoke_fn tag_invoke{};
                                            ^
```

**To Reproduce**
```c++
#include <dplx/cncr/tag_invoke.hpp>

namespace dplx::cncr
{
struct ti_test_type
{
    std::size_t providedN{};
    bool called{};
};

inline constexpr struct ti_test_cp_fn
{
    template <typename T>
        requires cncr::tag_invocable<ti_test_cp_fn, T &, std::size_t>
    constexpr decltype(auto) operator()(T &self, std::size_t n) const
    {
        return cncr::tag_invoke(*this, self, n);
    }
} ti_test_cp{};

auto tag_invoke(ti_test_cp_fn, cncr_tests::ti_test_type &self, std::size_t n)
        -> cncr_tests::ti_test_type &
{
    self.providedN = n;
    self.called = true;
    return self;
}
} // namespace dplx::cncr
```

**Expected behavior**
It should be possible to define and implement `tag_invoke` based CPOs within the `cncr` namespace.

**Environment**
 - Project Version: 1f4079b7407e94a426990dfe94c5b65c31ef2f85

--- a/CMakePresets.json
+++ b/CMakePresets.json
@@ -138,6 +138,7 @@
                             ".git",
                             ".vs",
                             ".vscode",
+                            "build",
                             "out",
                             "vcpkg_installed"
                         ]

--- a/src/cncr_tests/test_utils.hpp
+++ b/src/cncr_tests/test_utils.hpp
@@ -7,6 +7,12 @@
 
 #pragma once
 
+#include <dplx/predef/compiler.h>
+
+#ifdef DPLX_COMP_GNUC_AVAILABLE
+#pragma GCC diagnostic ignored "-Wmissing-declarations"
+#endif
+
 namespace dplx
 {
 }

--- a/src/dplx/cncr/tag_invoke.hpp
+++ b/src/dplx/cncr/tag_invoke.hpp
@@ -44,23 +44,23 @@ using tag_t = std::decay_t<decltype(Tag)>;
 
 inline namespace cpo
 {
-inline constexpr detail::cpo::tag_invoke_fn tag_invoke{};
+inline constexpr dplx::detail::cpo::tag_invoke_fn tag_invoke{};
 }
 
 template <typename Tag, typename... TArgs>
 concept tag_invocable
-        = std::invocable<detail::cpo::tag_invoke_fn, Tag, TArgs...>;
+        = std::invocable<dplx::detail::cpo::tag_invoke_fn, Tag, TArgs...>;
 
 template <typename Tag, typename... TArgs>
 concept nothrow_tag_invocable = tag_invocable<Tag, TArgs...> && std::
-        is_nothrow_invocable_v<detail::cpo::tag_invoke_fn, Tag, TArgs...>;
+        is_nothrow_invocable_v<dplx::detail::cpo::tag_invoke_fn, Tag, TArgs...>;
 
 template <typename Tag, typename... TArgs>
 using tag_invoke_result
-        = std::invoke_result<detail::cpo::tag_invoke_fn, Tag, TArgs...>;
+        = std::invoke_result<dplx::detail::cpo::tag_invoke_fn, Tag, TArgs...>;
 
 template <typename Tag, typename... TArgs>
 using tag_invoke_result_t
-        = std::invoke_result_t<detail::cpo::tag_invoke_fn, Tag, TArgs...>;
+        = std::invoke_result_t<dplx::detail::cpo::tag_invoke_fn, Tag, TArgs...>;
 
 } // namespace dplx::cncr

--- a/src/dplx/cncr/tag_invoke.hpp
+++ b/src/dplx/cncr/tag_invoke.hpp
@@ -8,10 +8,9 @@
 #pragma once
 
 #include <concepts>
-
 #include <type_traits>
 
-namespace dplx::detail::cpo_impl
+namespace dplx::detail::cpo
 {
 
 struct tag_invoke_fn
@@ -35,7 +34,7 @@ struct tag_invoke_fn
 
 void tag_invoke() = delete;
 
-} // namespace dplx::detail::cpo_impl
+} // namespace dplx::detail::cpo
 
 namespace dplx::cncr
 {
@@ -43,22 +42,25 @@ namespace dplx::cncr
 template <auto &Tag>
 using tag_t = std::decay_t<decltype(Tag)>;
 
-inline constexpr detail::cpo_impl::tag_invoke_fn tag_invoke{};
+inline namespace cpo
+{
+inline constexpr detail::cpo::tag_invoke_fn tag_invoke{};
+}
 
 template <typename Tag, typename... TArgs>
 concept tag_invocable
-        = std::invocable<detail::cpo_impl::tag_invoke_fn, Tag, TArgs...>;
+        = std::invocable<detail::cpo::tag_invoke_fn, Tag, TArgs...>;
 
 template <typename Tag, typename... TArgs>
 concept nothrow_tag_invocable = tag_invocable<Tag, TArgs...> && std::
-        is_nothrow_invocable_v<detail::cpo_impl::tag_invoke_fn, Tag, TArgs...>;
+        is_nothrow_invocable_v<detail::cpo::tag_invoke_fn, Tag, TArgs...>;
 
 template <typename Tag, typename... TArgs>
 using tag_invoke_result
-        = std::invoke_result<detail::cpo_impl::tag_invoke_fn, Tag, TArgs...>;
+        = std::invoke_result<detail::cpo::tag_invoke_fn, Tag, TArgs...>;
 
 template <typename Tag, typename... TArgs>
 using tag_invoke_result_t
-        = std::invoke_result_t<detail::cpo_impl::tag_invoke_fn, Tag, TArgs...>;
+        = std::invoke_result_t<detail::cpo::tag_invoke_fn, Tag, TArgs...>;
 
 } // namespace dplx::cncr

--- a/src/dplx/cncr/tag_invoke.test.cpp
+++ b/src/dplx/cncr/tag_invoke.test.cpp
@@ -29,8 +29,9 @@ struct ti_test_type
     std::size_t providedN{};
     bool called{};
 
-    friend inline auto tag_invoke(ti_test_cp_fn, ti_test_type &self, std::size_t n)
-            -> ti_test_type &
+    friend inline auto tag_invoke(ti_test_cp_fn,
+                                  ti_test_type &self,
+                                  std::size_t n) -> ti_test_type &
     {
         self.providedN = n;
         self.called = true;
@@ -38,10 +39,78 @@ struct ti_test_type
     }
 };
 
+} // namespace cncr_tests
+
+namespace dplx::detail
+{
+
+// test tag_invoke poison pill
+auto tag_invoke(cncr_tests::ti_test_cp_fn,
+                cncr_tests::ti_test_type &self,
+                std::size_t n) -> cncr_tests::ti_test_type &
+{
+    self.providedN = n;
+    self.called = false;
+    return self;
+}
+
+} // namespace dplx::detail
+
+namespace dplx::cncr
+{
+
+struct ti_test_type
+{
+    std::size_t providedN{};
+    bool called{};
+
+    friend inline auto tag_invoke(cncr_tests::ti_test_cp_fn,
+                                  ti_test_type &self,
+                                  std::size_t n) -> ti_test_type &
+    {
+        self.providedN = n;
+        self.called = true;
+        return self;
+    }
+};
+
+inline constexpr struct ti_test_cp_fn
+{
+    template <typename T>
+        requires cncr::tag_invocable<ti_test_cp_fn, T &, std::size_t>
+    constexpr decltype(auto) operator()(T &self, std::size_t n) const
+    {
+        return cncr::tag_invoke(*this, self, n);
+    }
+} ti_test_cp{};
+
+// test tag_invoke definitions within dplx::cncr
+auto tag_invoke(ti_test_cp_fn, cncr_tests::ti_test_type &self, std::size_t n)
+        -> cncr_tests::ti_test_type &
+{
+    self.providedN = n;
+    self.called = true;
+    return self;
+}
+
+} // namespace dplx::cncr
+
+namespace cncr_tests
+{
 TEST_CASE("tag_invoke should dispatch")
 {
     ti_test_type testInstance;
     ti_test_type &testRef = ti_test_cp(testInstance, 13U);
+
+    CHECK(testInstance.called);
+    CHECK(testInstance.providedN == 13U);
+    CHECK(&testInstance == &testRef);
+}
+
+TEST_CASE("tag_invoke should dispatch to cncr namespace declarations")
+{
+    ti_test_type testInstance;
+    ti_test_type &testRef = cncr::ti_test_cp(testInstance, 13U);
 
     CHECK(testInstance.called);
     CHECK(testInstance.providedN == 13U);

